File: bench/python/06-results-patterns/03-destructuring.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

def describe(point):
    match (point.x, point.y):
        case (0, 0):
            print("origin")
        case (x, 0):
            print("x axis")
            print(x)
        case (x, y):
            print(x + y)

File: bench/rust/06-results-patterns/03-destructuring.rs
struct Point { x: i64, y: i64 }
fn describe(point: Point) {
    match point { Point { x: 0, y: 0 } => println!("origin"), Point { x, y: 0 } => println!("x axis\n{x}"), Point { x, y } => println!("{}", x + y) }
}
